cylc/flow/task_job_mgr.py

```diff
@@ -996,23 +998,33 @@ def _set_retry_timers(
 
     def _simulation_submit_task_jobs(self, itasks, workflow):
         """Simulation mode task jobs submission."""
+        now = time()
+        now_str = get_time_string_from_unix_time(now)
         for itask in itasks:
+            itask.summary['started_time'] = now
+            self._set_retry_timers(itask, itask.tdef.rtconfig)
+            itask.mode_settings = ModeSettings(
+                itask, self.task_events_mgr.broadcast_mgr)
```

Comment:
But I'm using the presence of the db manager as part of the test for the workflow being restarted!

Reply:
Not if the `mode_settings` object is being created by the first job submission as is happening here. I think you are right though, it'd make sense to have an explicit flag rather than using the presence of the db_mgr object as an implicit flag, which is what I'm doing.

Reply:
Yes.